### DISCUSSION TITLE
Is it possible to only disassemble functions at specific address?

### DISCUSSION THREAD

**Discussion by ChiChou** [Vector35/binaryninja-api#4280] [API] [answered]
I only need to get the disassembly and control flow of some functions.

I tried `analysis.initialAnalysisHold` option, then

```python
for f in bv.functions:
    bv.remove_function(f)

bv.add_function(0x100001234)
bv.set_analysis_hold(False)
```

However as soon as I disable analysis hold, all of the functions come back. I think it might processes `LC_FUNCTION_STARTS` again discarding my settings. Is it possible to force Binary Ninja to only disassemble some functions?

**Comment by bpotchik** [comment, marked as answer]
This is possible however the analysis hold must be left in the enabled state during the entire process.

```
>>> bv.analysis_progress
<progress: Hold>
>>> func = bv.get_function_at(here)
>>> len(func.basic_blocks)
0
>>> func.request_advanced_analysis_data()
>>> len(func.basic_blocks)
5
```

With the analysis hold enabled, the top level analysis pipeline remains in the hold state and no other functions are analyzed. This allows manual analysis of a small subset of the binary. It's worth noting that disassembly, LLIL, MLIL, and HLIL are all generated for the function in the default analysis mode. By adjusting the `analysis.mode` setting, it is possible to limit the single function analysis to just disassembly, or have it stop at one of the ILs.

**Reply by ChiChou** [reply]
This is exactly what I need. Thank you so much!
